### DISCUSSION TITLE
Architecture code auto-generation

### DISCUSSION THREAD

**Discussion by downdawn** [fastapi-practices/fastapi_best_architecture#106] [Ideas]
To develop a new interface, you first need to define the model (migration) and schema, and then define crud, service and router (view), referred to as mscsr.

Among them, crud has the CRUDBase class, which supports common get, create, update, and delete methods.

Idea one:
So can the service layer also extract the ServiceBase class and support common methods, such as: query, create, update, delete (logical delete and true delete).

Idea two:
Can we support a command line tool that creates the basic structure mscsr with one line of command. for example:
```
fba create -m user
```

In addition, the readme document may need to supplement the pseudo-three-tier architecture and how to use it quickly.

**Comment by wu-clan** [comment]
This sounds like a daunting task, but I am very interested.

1. ServiceBase is not possible, services are not as generic as CRUD
2. This should not be a CLI, but a template service

Pseudo three-tier architecture:

Yes, we are now using a three-tier architecture, which requires some clarification as to why it is a pseudo three-tier architecture

**Comment by wu-clan** [comment]
related #318
